### DISCUSSION TITLE
Impulse v2.1.8

### DISCUSSION THREAD

**Discussion by ssesselmann** [ssesselmann/impulse#43] [Ideas]
**New in version 2.1.8**

1) Added new variable for lost counts (audio spectrometry only). This field shows how many counts are lost as a result of the pulse shape being outside the set tolerance. This field is also recorded in the json file. 

2) Moved all css styling from a separately hosted web site to the app bundle.

Bundled apps for Windows (impulse.zip) and Macintosh (impulse.dmg) can be downloaded below.

Note: I rely on people like you to make impulse better, so please report all bugs. 

<hr /><em>This discussion was created from the release <a href='https://github.com/ssesselmann/impulse/releases/tag/v2.1.8'>Impulse v2.1.8</a>.</em>

**Comment by jbeale1** [comment]
Sorry if this is out of place, but is there any possibility of running something like Impulse on a Raspberry Pi? I was thinking no GUI at all, just some command-line thing that writes out the count rate and accumulated spectrum as data files, after some time interval.

**Reply by BOPOHOP** [reply]
I haven't tried version 2, but version 1 worked on my  raspberry pi4b with both audio and atomspectra (*-max-*) spectrometers.

**Reply by ssesselmann** [reply]
Yes, you should be able to run Impulse on Linux an RPI, you just need to install Python and import the libraries.
If you are running it with a sound card spectrometer from the command line you only need some of the few files, most of it happens in 

impulse.py
functions.py
global_vars.py
launcher.py
server.py
shapecatcher.py
pulsecatcher.py
global_vars.py
functions.py

I think that's all, but without a browser you need to add a few lines of code to start and stop the functions.

Steven

**Comment by BOPOHOP** [comment]
You are speaking about running impulse without any UI. To do it we need add autostart of  spectrum collection. For version 1 start code is really simple
```
import functions as fn
import pulsecatcher as pc
pc.pulsecatcher(2)
```

Raspberry Pi4b can run "fully functional" impulse. We just need to separate browser. 
To do it   comment out /delete/ line "Timer(1, lambda: open_browser(port)).start()" from impulse.py, run "python3 impulse.py"  and connect from  desktop browser to http://RasprerryPiHostNameOrIipAddress:8050

**Reply by ssesselmann** [reply]
Because audio spectrometry uses the mean pulse shape from your detector to filter pulses, you also need a shape file, this can be obtained by running shapecatcher first.

Typically you want to get the mean shape from around 500-1000 pulses, this is referred to as 'shapecatches" in global_vars.py

shapecatches    = 1000

Once this number has been reached it writes "shape.csv" , the shape should remain unchanged as long as you use the same detector.

import functions as fn
import pulsecatcher as pc
import shapecatcher as sc
sc.shapecatcher()
pc.pulsecatcher(2)
